misc/pytorch_toolkit/chest_xray_screening/tools/train.py

```diff
@@ -0,0 +1,280 @@
+            'best_loss': valid_loss_min,
+            'optimizer' : self.optimizer.state_dict()},
+            savepath+'/m-epoch'+str(epoch_id)+'-' + timestamp_launch + '.pth.tar')
+            test_auroc = RSNATrainer.test(self)
+            print(f"Epoch:{epoch_id + 1}| EndTime:{timestamp_end}| TestAUROC: {test_auroc}| ValidAUROC: {auroc_max}")
+
+    def valid(self):
+        self.model.eval()
+        loss_valid_r = 0
+        valid_batches = 0      # Counter for valid batches
+        out_gt = torch.FloatTensor().to(self.device)
+        out_pred = torch.FloatTensor().to(self.device)
+        with torch.no_grad():
+            for (var_input, var_target) in tq(self.data_loader_valid):
+                # print(f"Batch {i} in Val")
+                var_target = var_target.to(self.device)
+                out_gt = torch.cat((out_gt, var_target), 0).to(self.device)
+
+                _, c, h, w = var_input.size()
+                var_input = var_input.view(-1, c, h, w)
+
+                var_output = self.model(var_input.to(self.device))
+                out_pred = torch.cat((out_pred, var_output), 0)
+
+                lossvalue = self.loss_fn(
+                    var_output,tfunc.one_hot(var_target.squeeze(1).long(),num_classes =self.class_count).float())
+
+                loss_valid_r += lossvalue.item()
+                valid_batches += 1
+
+            valid_loss = loss_valid_r / valid_batches
+
+            auroc_individual = compute_auroc(
+                tfunc.one_hot(out_gt.squeeze(1).long()).float(),
+                out_pred, self.class_count)
+            auroc_mean = np.array(auroc_individual).mean()
+        return valid_loss,auroc_mean
+
+
+    def epoch_train(self,savepath):
+        auroc_max = 0.0 # Setting maximum AUROC value as zero
+        if self.model_checkpoint is not None:
+            self.optimizer.load_state_dict(self.model_checkpoint['optimizer'])
+        loss_train_list = []
+        loss_valid_list = []
+        self.model.train()
+        scheduler = StepLR(self.optimizer, step_size=6, gamma=0.002)
+
+        for batch_id, (var_input, var_target) in tq(enumerate(self.data_loader_train)):
+            epoch_id = self.gepoch_id
+            var_target = var_target.to(self.device)
+            var_input = var_input.to(self.device)
+            var_output= self.model(var_input)
+            trainloss_value = self.loss_fn(
+                var_output,
+                tfunc.one_hot(var_target.squeeze(1).long(),num_classes=self.class_count).float())
+
+            self.optimizer.zero_grad()
+            trainloss_value.backward()
+            self.optimizer.step()
+            train_loss_value = trainloss_value.item()
+            loss_train_list.append(train_loss_value)
+
+            # Evaluate the performance of the model on validation_set
+            # every 2500th iteration. 2500 is a random choice, this could be changed.
+
+            if batch_id%2500==0 and batch_id!=0:
```

Comment:
[Magic number](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad). Declare a constant in the head of the file (between the imports and the `class` statement), set it 2500 and refer to it.